src/segment_battery.go

```diff
@@ -34,7 +34,7 @@ const (
 
 func (b *batt) enabled() bool {
 	batteries, err := b.env.getBatteryInfo()
-	if !b.enabledWhileError(err) {
+	if !b.enabledWhileError(err) || len(batteries) == 0 {
```

Comment:
I did mis this when refactoring. Should be good (although the tests probably don't cover the use-case)

Reply:
@lnu can you add a tests that ensures this is validated as well?